src-tauri/src/utils.rs

```diff
@@ -13,13 +34,11 @@ pub async fn fetch_services_manifests(url: &str, state: Arc<SharedState>) -> Res
         .await
         .with_context(|| "Failed to parse response to list of services")?;
     let mut services_guard = state.services.lock().await;
-    // TODO: discuss why do we need global ids, why not use uuids and generate them on each load
-    *services_guard = services
-        .into_iter()
-        // removes services without id
-        .filter_map(|x| Some((x.id.clone()?, x)))
-        // removes duplicate services
-        .collect();
+    for service in services {
+        if !services_guard.contains_key(&service.id.clone().unwrap_or_default()) {
+            services_guard.insert(service.get_id()?, service);
+        }
+    }
```

Comment:
```suggestion
        if !service.get_id_ref().map(|id| services_guard.contains_key(id)).unwrap_or_default() {
            services_guard.insert(service.get_id()?, service);
        }
    }
```
No need to search if no id found.
Also, any reasons to use different ways of getting id in both cases?